FrequentFlyer/Jobs/JobsDataDeserializer.swift

```diff
@@ -5,50 +5,28 @@ class JobsDataDeserializer {
     var buildDataDeserializer = BuildDataDeserializer()
 
     func deserialize(_ data: Data) -> Observable<[Job]> {
-        var jobsJSONObject: Any?
-        do {
-            jobsJSONObject = try JSONSerialization.jsonObject(with: data, options: .allowFragments)
-        } catch { }
-
-        var jobs = [Job]()
+        let jobsJSONObject = try? JSONSerialization.jsonObject(with: data, options: .allowFragments)
 
         guard let jobsJSON = jobsJSONObject as? Array<NSDictionary> else {
             return Observable.error(DeserializationError(details: "Could not interpret data as JSON dictionary", type: .invalidInputFormat))
         }
-
-        for jobsDictionary in jobsJSON {
-            guard let name = jobsDictionary["name"] as? String else { continue }
-            guard let groups = jobsDictionary["groups"] as? Array<String> else { continue }
-
-            let finishedBuildJSON = jobsDictionary["finished_build"] as? NSDictionary
-            let nextBuildJSON = jobsDictionary["next_build"] as? NSDictionary
-
-            if finishedBuildJSON == nil && nextBuildJSON == nil { continue }
-
-            var finishedBuild: Build? = nil
-            if let finishedBuildJSON = finishedBuildJSON {
-                do {
-
-                    let finishedBuildData = try JSONSerialization.data(withJSONObject: finishedBuildJSON, options: .prettyPrinted)
-                    finishedBuild = buildDataDeserializer.deserialize(finishedBuildData).build
-                } catch { }
-            }
-
-            var nextBuild: Build? = nil
-            if let nextBuildJSON = nextBuildJSON {
-                do {
-
-                    let nextBuildData = try JSONSerialization.data(withJSONObject: nextBuildJSON, options: .prettyPrinted)
-                    nextBuild = buildDataDeserializer.deserialize(nextBuildData).build
-                } catch { }
-            }
-
-
-            jobs.append(
-                Job(name: name, nextBuild: nextBuild, finishedBuild: finishedBuild, groups: groups)
-            )
+        
+        let jobs = jobsJSON.flatMap { jobsDictionary -> Job? in
+            guard let name = jobsDictionary["name"] as? String else { return nil }
+            guard let groups = jobsDictionary["groups"] as? Array<String> else { return nil }
+            
+            guard let finishedBuildJSON = jobsDictionary["finished_build"] as? NSDictionary else { return nil }
+            guard let nextBuildJSON = jobsDictionary["next_build"] as? NSDictionary else { return nil }
```

Comment:
I noticed after posting this that I've slightly modified the behaviour here. The original code would omit the job only if there were no finished build or next build, whereas this aborts if either are missing. The tests don't seem to mind, but I don't know if this is an important case that isn't covered.